RecoLocalCalo/HcalRecProducers/src/HcalRecHitSoAToLegacy.cc

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include <string>
+
+#include "DataFormats/HcalRecHit/interface/HcalRecHitHostCollection.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+
+class HcalRecHitSoAToLegacy : public edm::stream::EDProducer<> {
+public:
+  explicit HcalRecHitSoAToLegacy(edm::ParameterSet const& ps);
+  ~HcalRecHitSoAToLegacy() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  void produce(edm::Event&, edm::EventSetup const&) override;
+
+private:
+  const bool produceLegacy_;
+
+  using IProductType = hcal::RecHitHostCollection;
+  const edm::EDGetTokenT<IProductType> recHitsM0TokenIn_;
+
+  const edm::EDPutTokenT<HBHERecHitCollection> recHitsLegacyTokenOut_;
+};
+
+void HcalRecHitSoAToLegacy::fillDescriptions(edm::ConfigurationDescriptions& confDesc) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("recHitsM0LabelIn", edm::InputTag{"hbheRecHitProducerPortable"});
+  desc.add<std::string>("recHitsLegacyLabelOut", "");
+  desc.add<bool>("produceLegacy", true);
+
+  confDesc.addWithDefaultLabel(desc);
+}
+
+HcalRecHitSoAToLegacy::HcalRecHitSoAToLegacy(const edm::ParameterSet& ps)
+    : produceLegacy_{ps.getParameter<bool>("produceLegacy")},
+      recHitsM0TokenIn_{consumes<IProductType>(ps.getParameter<edm::InputTag>("recHitsM0LabelIn"))},
+      recHitsLegacyTokenOut_{produceLegacy_
+                                 ? produces<HBHERecHitCollection>(ps.getParameter<std::string>("recHitsLegacyLabelOut"))
```

Comment:
Is this label needed ?
This `EDProducer` produces only one collection, so the label seems superfluous ?

Reply:
Done in https://github.com/cms-sw/cmssw/commit/73cdabc345156cfa7129330710e3a023ca9a6404